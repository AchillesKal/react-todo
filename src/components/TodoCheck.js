import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheckCircle, faCircle } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';

function TodoCheck(props) {

  const status = props.status;
  if (status) {
    return <FontAwesomeIcon icon={faCheckCircle} />;
  }
  return <FontAwesomeIcon icon={faCircle} />;
};

TodoCheck.propTypes = {
  status: PropTypes.bool.isRequired
};

export default TodoCheck;