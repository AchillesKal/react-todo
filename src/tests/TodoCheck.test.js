import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TodoCheck from '../components/TodoCheck';

describe('TodoCheck Component', () => {
  it("Test checkbox is not checked", () => {
    const tree = renderer.create(<TodoCheck status={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Test checkbox is checked", () => {
    const tree = renderer.create(<TodoCheck status={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});