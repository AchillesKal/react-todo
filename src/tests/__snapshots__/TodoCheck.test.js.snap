// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCheck Component Test checkbox is checked 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-check-circle fa-w-16 "
  data-icon="check-circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`TodoCheck Component Test checkbox is not checked 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle fa-w-16 "
  data-icon="circle"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
